forest/src/daemon.rs

```diff
@@ -36,11 +36,23 @@ use wallet::{KeyStore, PersistentKeyStore};
 const WORKER_TASKS: usize = 1;
 
 /// Import a chain from a CAR file
-async fn import_chain<V: ProofVerifier, R: Read, DB: BlockStore>(
+async fn import_chain<V: ProofVerifier, DB: BlockStore>(
     bs: Arc<DB>,
-    reader: R,
+    path: String,
     snapshot: bool,
 ) -> Result<(), Box<dyn std::error::Error>> {
+    let is_remote_file: bool = path.starts_with("http://") || path.starts_with("https://");
+
+    let mut file_path = path;
+    if is_remote_file {
+        match download_file(file_path).await {
+            Ok(file) => file_path = file,
+            Err(err) => return Err(err),
+        }
+    }
```

Comment:
There is no need to save the file to file storage before loading it, it's completely unnecessary and will be very inefficient with very large car files. Should just load data received from network as the reader for `load_car`

Reply:
good point. My intent originally was to have a generic function that will be used for file downloading. However it's better to just return a `Reader`. Updated now.